### ISSUE TITLE
yay tries to pull ARM-only dependencies and fails

### ISSUE THREAD

**Issue by pavelxdd** [Jguer/yay#1953]
### Affected Version

yay v11.3.1.r126.gc744058 - libalpm v13.0.2

### Describe the bug

It's impossible to build [repman-git](https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=repman-git) package from AUR via yay on x86_64 right now.

### Reproduction Steps

1. `yay -S repman-git`

### Expected behavior

`devtools-alarm` is an ARM-specific dependency of `repman-git`:
https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=repman-git#n25
and shouldn't be pulled by yay.

### Output

```sh
❯ yay -S repman-git
 -> No AUR package found for devtools-alarm
 -> could not find all required packages: devtools-alarm
```

**Comment by Jguer** [comment]
https://aur.archlinux.org/rpc?arg[]=repman-git&type=info&v=5

The AUR RPC does not distinguish between arm and x86 deps. Since it's not possible to find it locally as well it's not possible to tell if it's this issue or a real missing dep.

Happy to accept ideas on how to get around this

**Comment by pavelxdd** [comment]
I'm not sure how all this done internally, can't you parse PKGBUILDs to filter out foreign architecture dependencies?

**Comment by grandchild** [comment]
The dependencies don't tell us whether they're for/from another architecture. It just says "devtools-alarm".
If your next thought is "let's filter out dependencies whose name matches `.*\balarm\b.*`" then that won't work either, because it's just not rigorous enough: Plenty of packages related to timers/notifications carry that name and a few packages (`distccd-*`) are even named after the architecture and _are_ valid x86_64 packages. On the other hand, there might be alarm- or i686 packages that _don't_ carry that in their names.
The fact that it's missing is also not enough, it might be a genuinely missing package.
If we wanted to start a block list of "known" non-x86_64 packages, then that's probably also a bad idea, because I can absolutely imagine a `devtools-alarm` package (possibly with very different contents, perhaps related to cross-compilation) to appear tomorrow.

In conclusion: Archlinux and especially the AUR just does not support different architectures at all well. :/ I suppose you're stuck with either:
- Deleting the dependency locally from the `repman-git` PKGBUILD before installing (`yay` helps you with that if you make local git commit of it, by rebasing? merging? it into the next update).
- Creating an empty local `devtools-alarm` PKGBUILD and package with that name and installing it, just to satisfy this package's dependencies.

**Comment by grandchild** [comment]
What you could also do is let the authors of `repman-git` know and ask them to (in the PKGBUILD) add `devtools-alarm` to the dependency list _only_ if they discover that the build is running on alarm (yes this is possible).

**Comment by pavelxdd** [comment]
Thanks for detailed explanation.

I've managed to install the package via `yay -G repman-git` + `makepkg -si`, not a big problem.

Should we close this issue then?

**Comment by grandchild** [comment]
Let's :)
